src/marqo/core/index_management/index_management.py

```diff
@@ -158,51 +182,62 @@ def batch_create_indexes(self, marqo_index_requests: List[MarqoIndexRequest]) ->
         Raises:
             IndexExistsError: If an index already exists
             InvalidVespaApplicationError: If Vespa application is invalid after applying the indexes
+            RuntimeError: If deployment lock is not instantiated
+            IndexCreationAndDeletionConflictError: If another index creation/deletion operation is
+                in progress and the lock cannot be acquired
         """
-        app = self.vespa_client.download_application()
-        configured = self._add_marqo_config(app)
+        if self._zookeeper_deployment_lock is None:
+            raise RuntimeError("Deployment lock is not instantiated and cannot be used for index creation/deletion")
```

Comment:
This check and error is repeated a number of times. Better to wrap it in an internal function and call that. You could even have a custom context manager (internal one) that wraps the lock context manager and performs this check as well. Both options fine

Reply:
Done